pkg/plugin/processor/builtin/js/processor.go

```diff
@@ -0,0 +1,370 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package js
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/dop251/goja"
+	"github.com/dop251/goja_nodejs/require"
+)
+
+const entrypoint = "process"
+
+// singleRecord is an intermediary representation of opencdc.Record that is passed to
+// the JavaScript transform. We use this because using opencdc.Record would not
+// allow us to modify or access certain data (e.g. metadata or structured data).
+type singleRecord struct {
+	Position  []byte
+	Operation string
+	Metadata  map[string]string
+	Key       any
+	Payload   struct {
+		Before any
+		After  any
+	}
+}
+
+type filterRecord struct {
+}
+
+type errorRecord struct {
+	Error string
+}
+
+// gojaContext represents one independent goja context.
+type gojaContext struct {
+	runtime  *goja.Runtime
+	function goja.Callable
+}
+
+type processor struct {
+	sdk.UnimplementedProcessor
+
+	// src is the JavaScript code that will be executed
+	src string
+
+	gojaPool sync.Pool
+	logger   log.CtxLogger
+}
+
+func New(logger log.CtxLogger) sdk.Processor {
+	return &processor{logger: logger}
+}
+
+func (p *processor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:        "custom.javascript",
+		Summary:     "JavaScript processor",
+		Description: "A JavaScript processor for Conduit",
+		Version:     "v0.1.0",
+		Author:      "Meroxa, Inc.",
+		Parameters: map[string]config.Parameter{
```

Comment:
can you please use paramgen for the params

Reply:
I was thinking about that... It didn't look much better with it, given that it's only 2 parameters, and that there's custom validation involved. I can give it another try though.

Reply:
I'm with Maha on this one, I think using the same approach for all processors will make it easier to maintain them. Let's say we change something in the parameter specs or add a feature to paramgen down the road, then it's trivial to regenerate the code for all processors.

Reply:
For the record, we had a chat about this on Slack, and we will probably provide a method that validates and parses a configuration, to remove some of the boilerplate code. That should make using paramgen even more compelling, even in cases where there are just 1 or 2 configuration parameters.